tool/tsh/common/kube.go

```diff
@@ -1649,11 +1635,15 @@ func (c *kubeLoginCommand) accessRequestForKubeCluster(ctx context.Context, cf *
 	}
 	defer clt.Close()
 
+	predicate := ""
+	if !c.all && c.labels == "" && c.predicateExpression == "" {
+		predicate = makeDiscoveredNameOrNamePredicate(c.kubeCluster)
+	}
```

Comment:
should this be `if c.kubeCluster != ""`? if the current `if` check is intended, could you add a comment?

Reply:
when --labels or --query are used, it works the same way as `--all` did except with filtering.
Meaning it will update the kube config to include the matched clusters and optionally select one for the current context if `c.kubeCluster != ""`.

If none of `--all/--labels/--query` are given (like `tsh kube login some-name`), then it only updates the kube config for `some-name` and also sets the current context.

This mean when making an access request, it should request just `some-kube` when none of `--all, --labels, --query` are given. Otherwise, it will need to request all the matched clusters (this is prior behavior as well):
prior behavior was:
```
	expr := ""
	if !cf.ListAll {
		expr = fmt.Sprintf(`resource.metadata.name == "%s"`, tc.KubernetesCluster)
	}
```

all that being said, I can add a comment 🤣

Reply:
moved it into a func with a lengthy explanation godoc: f09ce2231fdfb17ce2f13c41d06863b5e8ef4058